File: solutions/php/phone-number/1/PhoneNumber.php
<?php

/*
 * By adding type hints and enabling strict type checking, code can become
 * easier to read, self-documenting and reduce the number of potential bugs.
 * By default, type declarations are non-strict, which means they will attempt
 * to change the original type to match the type specified by the
 * type-declaration.
 *
 * In other words, if you pass a string to a function requiring a float,
 * it will attempt to convert the string value to a float.
 *
 * To enable strict mode, a single declare directive must be placed at the top
 * of the file.
 * This means that the strictness of typing is configured on a per-file basis.
 * This directive not only affects the type declarations of parameters, but also
 * a function's return type.
 *
 * For more info review the Concept on strict type checking in the PHP track
 * <link>.
 *
 * To disable strict typing, comment out the directive below.
 */

declare(strict_types=1);

class PhoneNumber
{
    private $pn;

    public function __construct($phone) {
        $phone=preg_replace("/[ \(\).\-\+]+/","",$phone);
        if (strlen($phone)>11) throw new InvalidArgumentException('more than 11 digits');
        if (strlen($phone)==11 &&substr($phone,0,1)!='1') throw new InvalidArgumentException('11 digits must start with 1');
        if (preg_match("/[a-zA-Z]+/",$phone)) throw new InvalidArgumentException('letters not permitted');
        if (preg_match("/[@:!]+/",$phone)) throw new InvalidArgumentException('punctuations not permitted');
        if (strlen($phone)==11) $phone=substr($phone,1,10);
        if (strlen($phone)<10) throw new InvalidArgumentException('incorrect number of digits');
        if (substr($phone,0,1)=='0') throw new InvalidArgumentException('area code cannot start with zero');
        if (substr($phone,0,1)=='1') throw new InvalidArgumentException('area code cannot start with one');
        if (substr($phone,3,1)=='0') throw new InvalidArgumentException('exchange code cannot start with zero');
        if (substr($phone,3,1)=='1') throw new InvalidArgumentException('exchange code cannot start with one');
        $this->pn=$phone;
    }

    public function number(): string
    {
        return $this->pn;
    }
}

File: solutions/java/phone-number/1/src/main/java/PhoneNumber.java
public class PhoneNumber {
    private final static String wrongLengthExceptionMessage = "Number must be 10 or 11 digits";
    private final static String numberIs11DigitsButDoesNotStartWith1ExceptionMessage =
            "Can only have 11 digits if number starts with '1'";
    private final static String illegalCharacterExceptionMessage =
            "Illegal character in phone number. Only digits, spaces, parentheses, hyphens or dots accepted.";
    private final static String illegalAreaOrExchangeCodeMessage =
            "Illegal Area Or Exchange Code. Only 2-9 are valid digits";
    String enteredNumber;
    String actualNumber;

    PhoneNumber(String enteredNumber) {
        String numbers = "0123456789";
        String validphonenumberchars = numbers + " ()-.+";
        String areacodenumbers = numbers.substring(2,10);

        //System.out.println("enteredNumber="+enteredNumber);
        actualNumber = "";
        for (int i = 0; i < enteredNumber.length(); i++) {
            if (validphonenumberchars.indexOf(enteredNumber.substring(i,i+1))<0)
                throw new IllegalArgumentException(illegalCharacterExceptionMessage);
            if (numbers.indexOf(enteredNumber.substring(i,i+1))>=0){
                actualNumber += enteredNumber.substring(i,i+1);
            }
        }
        //System.out.println("actualNumber="+actualNumber);
        if (actualNumber.length() != 10 && actualNumber.length() != 11 )
            throw new IllegalArgumentException(wrongLengthExceptionMessage);
        if (actualNumber.length() == 11 && actualNumber.substring(0,1).equals("1") == false)
            throw new IllegalArgumentException(numberIs11DigitsButDoesNotStartWith1ExceptionMessage);
        if (actualNumber.length() == 11 && actualNumber.substring(0,1).equals("1"))
            actualNumber = actualNumber.substring(1,11);
        int j=0;
        if (actualNumber.length() == 11 && actualNumber.substring(0,1).equals("1")) j = 1;
        for (int i = 0; i < 3; i++) {
            //System.out.println("j="+j+",i="+i);
            //System.out.println(actualNumber.substring(j+i,j+i+1));
            //System.out.println(actualNumber.substring(j+i+3,j+i+1+3));
            if (areacodenumbers.indexOf(actualNumber.substring(j+i,j+i+1))<0)
                throw new IllegalArgumentException(illegalAreaOrExchangeCodeMessage);
            if (areacodenumbers.indexOf(actualNumber.substring(j+i+3,j+i+1+3))<0)
                throw new IllegalArgumentException(illegalAreaOrExchangeCodeMessage);
        }
    }

    String getNumber() {
        return(actualNumber);
    }
}